common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -1232,6 +1232,23 @@ func (d *cassandraPersistence) UpdateWorkflowExecution(
 func (d *cassandraPersistence) ConflictResolveWorkflowExecution(
 	request *p.InternalConflictResolveWorkflowExecutionRequest,
 ) error {
+	// first append history events
+	for _, req := range request.ResetWorkflowEventsNewEvents {
+		if err := d.AppendHistoryNodes(req); err != nil {
+			return err
+		}
+	}
+	for _, req := range request.NewWorkflowEventsNewEvents {
+		if err := d.AppendHistoryNodes(req); err != nil {
+			return err
+		}
+	}
+	for _, req := range request.CurrentWorkflowEventsNewEvents {
+		if err := d.AppendHistoryNodes(req); err != nil {
+			return err
+		}
+	}
+
```

Comment:
nit: can we do the current workflow events first, then reset, then new?

Reply:
sure

Reply:
what is the difference doing them in different order?

Reply:
persistence layer logic usually first touch the current workflow first, see ref: https://github.com/temporalio/temporal/pull/1926/files#diff-f5746d819e8ff4338478e5ecf595530831afb175cc000f7258b82135a2c744ffR1265